File: java/baek-joon/silver/15688-수 정렬하기5.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();

		int n = Integer.parseInt(br.readLine());
		int nums[] = new int[n];

		for (int i = 0; i < n; i++) {
			nums[i] = Integer.parseInt(br.readLine());
		}

		Arrays.sort(nums);

		for (int i = 0; i < n; i++) {
			sb.append(nums[i]).append("\n");
		}
		
		System.out.println(sb);
	}
}

// 같은 로직으로 kotlin을 사용했을 때 시간초과가 발생하여
// 자바로 해결

File: kotlin/baek-joon/silver/15688-수정렬하기5.kt
import java.io.BufferedWriter
import java.io.OutputStreamWriter
import java.util.*

val bw = BufferedWriter(OutputStreamWriter(System.out))
fun main() = with(System.`in`.bufferedReader()) {
    val n = readLine().toInt()
    val arr = Array<Int>(n){0}
    val sb = StringBuilder()
    for(i in 0 until n){
        arr[i] = Integer.parseInt(readLine())
    }

    Arrays.sort(arr)
    for(i in arr){
        sb.append(i).append("\n")
    }
    println(sb)

}
// 이 코드는 실제로 돌리면 시간 초과가 난다...
// 이 로직을 그대로 자바로 바꾸면 시간 초과가 해소된다..
// 코틀린은 생각보다 느리다..